core-foundation/src/array.rs

```diff
@@ -33,15 +34,47 @@ unsafe impl FromVoid for u32 {
     }
 }
 
-unsafe impl FromVoid for *const c_void {
-    unsafe fn from_void(x: *const c_void) -> *const c_void {
-        x
+unsafe impl<T> FromVoid for *const T {
+    unsafe fn from_void(x: *const c_void) -> Self {
+        x as *const T
     }
 }
 
-unsafe impl FromVoid for CFType {
-    unsafe fn from_void(x: *const c_void) -> CFType {
-        TCFType::wrap_under_get_rule(mem::transmute(x))
+unsafe impl<T: TCFType> FromVoid for T where T::Ref: FromVoid {
+    unsafe fn from_void(x: *const c_void) -> Self {
+        TCFType::wrap_under_get_rule(T::Ref::from_void(x))
```

Comment:
`TCFType::Ref` is already forced to implement `TCFTypeRef`, which has a `from_void_ptr` method. So you could remove `where T::Ref: FromVoid` and use `T::Ref::from_void_ptr(x)` instead. Would work the same, but would be slightly simpler/more minimal maybe?

Reply:
I went with this approach